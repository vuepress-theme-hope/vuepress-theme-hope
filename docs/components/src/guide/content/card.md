---
title: VPCard
---

Card component, can be use to display items.

<!-- more -->

## Demo

<!-- #region demo -->

::: md-demo Basic Card

<VPCard
  title="Mr.Hope"
  desc="Where there is light, there is hope"
  logo="https://mister-hope.com/logo.svg"
  link="https://mister-hope.com"
  background="rgba(253, 230, 138, 0.15)"
/>

:::

<!-- #endregion demo -->

## Props

### title

- Type: `string`
- Required: Yes

Card title

### desc

- Type: `string`
- Default: `''`

Card description

### logo

- Type: `string`
- Required: No

Card logo

### link

- Type: `string`
- Required: No

Card link

### background

- Type: `string`
- Required: No

Card background

### color

- Type: `string`
- Required: No

Card font color

::: tip

To make background and font color adapt to dark mode automatically, you can pass css variable, such as: `var(--my-bg)`.

:::

## Container

If you want multiple cards in a responsive container, you can wrap them in a `div` with class `vp-card-container`:

::: md-demo Responsive Card Container

<div class="vp-card-container">
  <VPCard
    v-for="i in 12"
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
  <VPCard
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
  <VPCard
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
</div>

:::
