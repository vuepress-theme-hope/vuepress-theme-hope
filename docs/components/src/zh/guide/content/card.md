---
title: VPCard
---

卡片组件，可用于展示项目。

<!-- more -->

## 案例

<!-- #region demo -->

::: md-demo 基础卡片

<VPCard
  title="Mr.Hope"
  desc="Where there is light, there is hope"
  logo="https://mister-hope.com/logo.svg"
  link="https://mister-hope.com"
  background="rgba(253, 230, 138, 0.15)"
/>

:::

<!-- #endregion demo -->

## 属性

### title

- 类型: `string`
- 必填: 是

卡片标题

### desc

- 类型: `string`
- Default: `''`

卡片描述

### logo

- 类型: `string`
- 必填: 否

卡片图标

### link

- 类型: `string`
- 必填: 否

卡片链接

### background

- 类型: `string`
- 必填: 否

卡片背景

### color

- 类型: `string`
- 必填: 否

卡片字体颜色

::: tip

为了让背景和字体颜色能自动适配夜间模式，你可以传入 css variable，如: `var(--my-bg)`。

:::

## 容器

如果你想要在一个响应式容器中放置多个卡片，你可以将它们包裹在一个 `div` 中，并添加 `vp-card-container` 类:

::: md-demo 响应式卡片容器

<div class="vp-card-container">
  <VPCard
    v-for="i in 12"
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
  <VPCard
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
  <VPCard
    title="Mr.Hope"
    desc="Where there is light, there is hope"
    logo="https://mister-hope.com/logo.svg"
    link="https://mister-hope.com"
    background="rgba(253, 230, 138, 0.15)"
  />
</div>

:::
