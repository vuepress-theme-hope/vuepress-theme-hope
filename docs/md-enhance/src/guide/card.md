---
title: Card
icon: square
---

You can add cards in Markdown content.

<!-- more -->

## Settings

::: code-tabs#language

@tab TS

```ts {8}
// .vuepress/config.ts
import { mdEnhancePlugin } from "vuepress-plugin-md-enhance";

export default {
  plugins: [
    mdEnhancePlugin({
      // Enable card support
      card: true,
    }),
  ],
};
```

@tab JS

```js {8}
// .vuepress/config.js
import { mdEnhancePlugin } from "vuepress-plugin-md-enhance";

export default {
  plugins: [
    mdEnhancePlugin({
      // Enable card support
      card: true,
    }),
  ],
};
```

:::

## Usage

You can use card fence block to add a card into your markdown content. Both YAML and JSON format card data are supported:

````md
```card
# card data here
```

```card
{
  // card data here
}
```
````

Card data supports `title`, `desc`, `logo`, `link` and `color` properties.

If you want to place multiple cards together, you can wrap them in `card` container:

````md
::: card

```card
# card data here
```

```card
# card data here
```

...

:::
````

## Demo

::: card

```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mister-hope.com/logo.svg
link: https://mister-hope.com
color: rgba(253, 230, 138, 0.15)
```

```card
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mister-hope.com/logo.svg",
  "link": "https://mister-hope.com",
  "color": "rgba(253, 230, 138, 0.15)"
}
```

:::

````md
```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mister-hope.com/logo.svg
link: https://mister-hope.com
color: rgba(253, 230, 138, 0.15)
```

```card
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mister-hope.com/logo.svg",
  "link": "https://mister-hope.com",
  "color": "rgba(253, 230, 138, 0.15)"
}
```
````
