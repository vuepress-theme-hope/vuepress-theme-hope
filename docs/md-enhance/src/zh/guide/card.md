---
title: 卡片
icon: square
---

你可以在 Markdown 中添加卡片。

<!-- more -->

## 配置

::: code-tabs#language

@tab TS

```ts {8}
// .vuepress/config.ts
import { mdEnhancePlugin } from "vuepress-plugin-md-enhance";

export default {
  plugins: [
    mdEnhancePlugin({
      // 开启卡片支持
      card: true,
    }),
  ],
};
```

@tab JS

```js {8}
// .vuepress/config.js
import { mdEnhancePlugin } from "vuepress-plugin-md-enhance";

export default {
  plugins: [
    mdEnhancePlugin({
      // 开启卡片支持
      card: true,
    }),
  ],
};
```

:::

## 使用

你可以使用卡片代码块来在 Markdown 中添加卡片。

````md
```card
# 卡片数据
```
````

默认情况下，你应该使用 YAML 格式来编写卡片数据，但你也可以通过在代码块后添加 `:json` 后缀来使用 JSON 格式。

````md
```card:json
{
  // 卡片数据
}
```
````

卡片数据支持 `title`、`desc`、`logo`、`link` 和 `color` 属性。

## 案例

```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mrhope.site/logo.svg
link: https://mrhope.site
color: rgba(253, 230, 138, 0.15)
```

```card:json
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mrhope.site/logo.svg",
  "link": "https://mrhope.site",
  "color": "rgba(253, 230, 138, 0.15)"
}
```

````md
```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mrhope.site/logo.svg
link: https://mrhope.site
color: rgba(253, 230, 138, 0.15)
```

```card:json
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mrhope.site/logo.svg",
  "link": "https://mrhope.site",
  "color": "rgba(253, 230, 138, 0.15)"
}
```
````
