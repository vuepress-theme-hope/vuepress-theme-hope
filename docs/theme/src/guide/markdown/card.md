---
title: Card
icon: square
category:
  - Markdown
tag:
  - Card
  - Markdown
---

You can add cards in Markdown content.

<!-- more -->

## Settings

::: code-tabs#language

@tab TS

```ts {8-10}
// .vuepress/config.ts
import { defineUserConfig } from "vuepress";
import { hopeTheme } from "vuepress-theme-hope";

export default defineUserConfig({
  theme: hopeTheme({
    plugins: {
      mdEnhance: {
        card: true,
      },
    },
  }),
});
```

@tab JS

```js {7-9}
// .vuepress/config.js
import { hopeTheme } from "vuepress-theme-hope";

export default {
  theme: hopeTheme({
    plugins: {
      mdEnhance: {
        card: true,
      },
    },
  }),
};
```

:::

## Usage

You can use card fence block to add a card into your markdown content. Both YAML and JSON format card data are supported:

````md
```card
# card data here
```

```card
{
  // card data here
}
```
````

Card data supports `title`, `desc`, `logo`, `link` and `color` properties.

If you want to place multiple cards together, you can wrap them in `card` container:

````md
::: card

```card
# card data here
```

```card
# card data here
```

...

:::
````

## Demo

::: card

```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mister-hope.com/logo.svg
link: https://mister-hope.com
color: rgba(253, 230, 138, 0.15)
```

```card
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mister-hope.com/logo.svg",
  "link": "https://mister-hope.com",
  "color": "rgba(253, 230, 138, 0.15)"
}
```

:::

````md
```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mister-hope.com/logo.svg
link: https://mister-hope.com
color: rgba(253, 230, 138, 0.15)
```

```card
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mister-hope.com/logo.svg",
  "link": "https://mister-hope.com",
  "color": "rgba(253, 230, 138, 0.15)"
}
```
````
