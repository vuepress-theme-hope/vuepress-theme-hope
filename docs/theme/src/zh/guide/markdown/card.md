---
title: 卡片
icon: square
category:
  - Markdown
tag:
  - Markdown
  - 对齐
---

你可以在 Markdown 中添加卡片。

<!-- more -->

## 配置

::: code-tabs#language

@tab TS

```ts {8-10}
// .vuepress/config.ts
import { defineUserConfig } from "vuepress";
import { hopeTheme } from "vuepress-theme-hope";

export default defineUserConfig({
  theme: hopeTheme({
    plugins: {
      mdEnhance: {
        card: true,
      },
    },
  }),
});
```

@tab JS

```js {7-9}
// .vuepress/config.js
import { hopeTheme } from "vuepress-theme-hope";

export default {
  theme: hopeTheme({
    plugins: {
      mdEnhance: {
        card: true,
      },
    },
  }),
};
```

:::

## 使用

你可以使用卡片代码块来在 Markdown 中添加卡片。

````md
```card
# 卡片数据
```
````

默认情况下，你应该使用 YAML 格式来编写卡片数据，但你也可以通过在代码块后添加 `:json` 后缀来使用 JSON 格式。

````md
```card:json
{
  // 卡片数据
}
```
````

卡片数据支持 `title`、`desc`、`logo`、`link` 和 `color` 属性。

## 案例

```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mrhope.site/logo.svg
link: https://mrhope.site
color: rgba(253, 230, 138, 0.15)
```

```card:json
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mrhope.site/logo.svg",
  "link": "https://mrhope.site",
  "color": "rgba(253, 230, 138, 0.15)"
}
```

````md
```card
title: Mr.Hope
desc: Where there is light, there is hope
logo: https://mrhope.site/logo.svg
link: https://mrhope.site
color: rgba(253, 230, 138, 0.15)
```

```card:json
{
  "title": "Mr.Hope",
  "desc": "Where there is light, there is hope",
  "logo": "https://mrhope.site/logo.svg",
  "link": "https://mrhope.site",
  "color": "rgba(253, 230, 138, 0.15)"
}
```
````
