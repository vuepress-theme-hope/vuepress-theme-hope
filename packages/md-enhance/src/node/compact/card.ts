import { container } from "@mdit/plugin-container";
import {
  entries,
  fromEntries,
  isPlainObject,
  isString,
} from "@vuepress/helper";
import { load } from "js-yaml";
import type { Options, PluginSimple } from "markdown-it";
import type Token from "markdown-it/lib/token.js";
import type { MarkdownEnv } from "vuepress/markdown";

import { stringifyProp } from "../markdown-it/utils.js";
import { logger } from "../utils.js";

export interface CardOptions {
  title: string;
  desc?: string;
  logo?: string;
  link?: string;
  color?: string;
}

const CARD_PROPS = ["title", "desc", "logo", "link", "color"];

const checkCardProps = (config: unknown): CardOptions | null => {
  if (isPlainObject(config) && isString(config["title"]))
    return fromEntries(
      entries(config).filter(
        (item): item is [string, string] =>
          CARD_PROPS.includes(item[0]) && isString(item[1]),
      ),
    ) as unknown as CardOptions;

  return null;
};

const cardRender = (
  tokens: Token[],
  index: number,
  _options: Options,
  { filePathRelative }: MarkdownEnv,
): string => {
  const token = tokens[index];
  const { content } = token;

  let config: unknown = null;

  if (content.trim().startsWith("{"))
    try {
      config = <unknown>JSON.parse(content);
    } catch (err) {
      // Do nothing
      logger.error(`Parsing card as JSON config failed:`, err);
    }
  else
    try {
      config = load(content);
    } catch (err) {
      logger.error(`Parsing card as YAML config failed:`, err);
    }

  const cardData = checkCardProps(config);

  if (cardData) return `<VPCard v-bind='${stringifyProp(cardData)}' />`;

  logger.error(
    `Invalid card config${
      filePathRelative ? ` found in ${filePathRelative}` : ""
    }:
${content}
`,
  );

  return "";
};

/** @deprecated */
export const legacyCard: PluginSimple = (md) => {
  // Add card container
  md.use(container, {
    name: "card",
    openRender: () =>
      `\
<div class="vp-card-container">
`,
  });

  // Handle ```card  blocks
  const { fence } = md.renderer.rules;

  md.renderer.rules.fence = (...args): string => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const [tokens, index, options, env] = args;
    const { info } = tokens[index];

    if (info === "card")
      return cardRender(tokens, index, options, <MarkdownEnv>env);

    const [realInfo] = info.split(":", 2);

    if (realInfo === "card") {
      logger.warn(
        "Language declaration for card is deprecated, please remove them.",
      );

      return cardRender(tokens, index, options, <MarkdownEnv>env);
    }

    return fence!(...args);
  };

  md.renderer.rules["card"] = cardRender;
};
