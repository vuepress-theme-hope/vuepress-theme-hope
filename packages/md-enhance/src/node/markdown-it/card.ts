import { type MarkdownEnv } from "@vuepress/markdown";
import { load } from "js-yaml";
import { type Options, type PluginSimple } from "markdown-it";
import type Token from "markdown-it/lib/token.js";
import {
  entries,
  fromEntries,
  isPlainObject,
  isString,
} from "vuepress-shared/node";

import { stringifyProp } from "./utils.js";
import { logger } from "../utils.js";

export interface CardOptions {
  title: string;
  desc?: string;
  logo?: string;
  link?: string;
  color?: string;
}

const CARD_PROPS = ["title", "desc", "logo", "link", "color"];

const checkCardProps = (config: unknown): CardOptions | null => {
  if (isPlainObject(config) && isString(config["title"]))
    return fromEntries(
      entries(config).filter(
        (item): item is [string, string] =>
          CARD_PROPS.includes(item[0]) && isString(item[1])
      )
    ) as unknown as CardOptions;

  return null;
};

const cardRender = (
  tokens: Token[],
  index: number,
  _options: Options,
  { filePathRelative }: MarkdownEnv
): string => {
  const token = tokens[index];
  const { content, info } = token;

  const language = info.trim().split(":", 2)[1] || "yml";
  let config: unknown = null;

  if (language === "yaml" || language === "yml") config = load(content);
  else if (language === "json")
    try {
      config = <unknown>JSON.parse(content);
    } catch (err) {
      // do nothing
    }

  if (!config) {
    logger.error(
      `Can not parse card config ${language}${
        filePathRelative ? `, found in ${filePathRelative}` : ""
      }.`
    );

    return "";
  }

  const cardData = checkCardProps(config);

  if (cardData) return `<VPCard v-bind='${stringifyProp(cardData)}' />`;

  logger.error(
    `Invalid card config${
      filePathRelative ? ` found in ${filePathRelative}` : ""
    }:
${content}
`
  );

  return "";
};

export const card: PluginSimple = (md) => {
  // Handle ```card  blocks
  const fence = md.renderer.rules.fence;

  md.renderer.rules.fence = (...args): string => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const [tokens, index, options, env] = args;
    const { info } = tokens[index];
    const realInfo = info.split(":", 2)[0];

    if (realInfo === "card")
      return cardRender(tokens, index, options, <MarkdownEnv>env);

    return fence!(...args);
  };

  md.renderer.rules["card"] = cardRender;
};
