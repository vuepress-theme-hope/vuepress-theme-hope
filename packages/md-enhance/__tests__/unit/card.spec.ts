import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { card } from "../../src/node/markdown-it/card.js";

describe("card", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(card);

  it("Should resolve card fence", () => {
    const result = markdownIt.render(
      `
\`\`\`card
title: A card
desc: A card desc
logo: https://example.com/logo.png
link: https://example.com
color: "#000"
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).toMatchSnapshot();
  });

  it("Should resolve card:yaml fence", () => {
    const result = markdownIt.render(
      `
\`\`\`card:yaml
title: A card
desc: A card desc
logo: https://example.com/logo.png
link: https://example.com
color: "#000"
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).toMatchSnapshot();
  });

  it("Should resolve card:yml fence", () => {
    const result = markdownIt.render(
      `
\`\`\`card:yml
title: A card
desc: A card desc
logo: https://example.com/logo.png
link: https://example.com
color: "#000"
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).toMatchSnapshot();
  });

  it("Should resolve card:json fence", () => {
    const result = markdownIt.render(
      `
\`\`\`card:json
{
  "title": "A card",
  "desc": "A card desc",
  "logo": "https://example.com/logo.png",
  "link": "https://example.com",
  "color": "#000"
}
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).toMatchSnapshot();
  });

  it("Should drop unknown keys", () => {
    const result = markdownIt.render(
      `
\`\`\`card:json
{
  "title": "A card",
  "desc": "A card desc",
  "logo": "https://example.com/logo.png",
  "link": "https://example.com",
  "color": "#000",
  "unknown": "value"
}
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).not.toContain("unknown");
    expect(result).not.toContain("value");
    expect(result).toMatchSnapshot();
  });

  it("Should drop invalid keys", () => {
    const result = markdownIt.render(
      `
\`\`\`card:json
{
  "title": "A card",
  "desc": "A card desc",
  "logo": 1
}
\`\`\`
`,
      {}
    );

    expect(result).toContain("VPCard");
    expect(result).not.toContain("logo");
    expect(result).toMatchSnapshot();
  });

  it("Should not throw with invalid syntax", () => {
    const result1 = markdownIt.render(
      `
\`\`\`card
title: a
title: b
\`\`\`
`,
      {}
    );

    const result2 = markdownIt.render(
      `
\`\`\`card:json
title: a
title: b
\`\`\`
`,
      {}
    );

    expect(result1).toEqual("");
    expect(result2).toEqual("");
  });

  it("Should not break markdown fence", () => {
    const result = markdownIt.render(
      `
\`\`\`js
const a = 1;
\`\`\`
`,
      {}
    );

    expect(result).toMatch(/<pre.*>[\s\S]*<\/pre>/);
    expect(result).toMatchSnapshot();
  });
});
