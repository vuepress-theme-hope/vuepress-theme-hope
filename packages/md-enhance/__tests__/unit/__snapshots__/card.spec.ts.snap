// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`card > Should drop invalid keys 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\"}' />"`;

exports[`card > Should drop unknown keys 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\",\\"logo\\":\\"https://example.com/logo.png\\",\\"link\\":\\"https://example.com\\",\\"color\\":\\"#000\\"}' />"`;

exports[`card > Should not break markdown fence 1`] = `
"<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
"
`;

exports[`card > Should resolve card fence 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\",\\"logo\\":\\"https://example.com/logo.png\\",\\"link\\":\\"https://example.com\\",\\"color\\":\\"#000\\"}' />"`;

exports[`card > Should resolve card:json fence 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\",\\"logo\\":\\"https://example.com/logo.png\\",\\"link\\":\\"https://example.com\\",\\"color\\":\\"#000\\"}' />"`;

exports[`card > Should resolve card:yaml fence 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\",\\"logo\\":\\"https://example.com/logo.png\\",\\"link\\":\\"https://example.com\\",\\"color\\":\\"#000\\"}' />"`;

exports[`card > Should resolve card:yml fence 1`] = `"<VPCard v-bind='{\\"title\\":\\"A card\\",\\"desc\\":\\"A card desc\\",\\"logo\\":\\"https://example.com/logo.png\\",\\"link\\":\\"https://example.com\\",\\"color\\":\\"#000\\"}' />"`;
